compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -3314,18 +3314,6 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
             format!("cannot check whether the hidden type of {name} satisfies auto traits"),
         );
         err.span_note(self.tcx.def_span(def_id), "opaque type is declared here");
-        match self.defining_use_anchor {
-            DefiningAnchor::Bubble | DefiningAnchor::Error => {}
-            DefiningAnchor::Bind(bind) => {
-                err.span_note(
-                    self.tcx.def_ident_span(bind).unwrap_or_else(|| self.tcx.def_span(bind)),
-                    "this item depends on auto traits of the hidden type, \
-                    but may also be registering the hidden type. \
-                    This is not supported right now. \
-                    You can try moving the opaque type and the item that actually registers a hidden type into a new submodule".to_string(),
-                );
-            }
-        };
```

Comment:
Not sure how to recover this. Maybe we could store information in the obligation cause code

Reply:
maybe simply stating "fetching the hidden types of an opaque inside of the defining scope is not supported" or sth like that is good enough?